src/compiler/crystal/interpreter/compiler.cr

```diff
@@ -2341,9 +2341,14 @@ class Crystal::Repl::Compiler < Crystal::Visitor
 
     request_value(arg)
 
-    # We first cast the argument to the def's arg type,
-    # which is the external methods' type.
-    downcast arg, arg_type, target_def_arg_type
+    # Check number autocast but for variables
```

Comment:
I presume this applies to constants as well as return values, which are all not variables?
```suggestion
    # Check number autocast but for non-literals
```